dynawo/sources/Models/CMakeLists.txt

```diff
@@ -294,12 +294,21 @@ macro(COMPILE_CPP_MODEL
       )
     target_link_libraries(${CPP_MODEL}
       PRIVATE
-        dynawo_Common
         dynawo_ModelerCommon
       PUBLIC
         dynawo_CPPModelsCommon
       )
 
+    if(LIBRARY_TYPE STREQUAL "STATIC")
+      target_include_directories(lib PRIVATE $<TARGET_PROPERTY:dynawo_Common,INTERFACE_INCLUDE_DIRECTORIES>)
+      if(MSVC OR APPLE)
+        # Link done to avoid duplicating the Trace singleton
+        target_link_libraries(${CPP_MODEL} PRIVATE dynawo_Common)
+      endif()
+    else()
+      target_link_libraries(${CPP_MODEL} PRIVATE dynawo_Common)
```

Comment:
you are still linking to dynawo_Common here

Reply:
Correct, there is an error. But in that case, I don't get to which target it is supposed to link (an executable ? another library ?)

Reply:
In shared you can link to the library containing the singleton (here that would still be dynawo_Common or dynawo_Util as I propose) and in static you need to link to the executable